java-api-core/src/com/kinvey/java/network/NetworkManager.java

```diff
@@ -740,15 +737,14 @@ public Aggregate aggregate(ArrayList<String> fields, AggregateType type, String
 
                 Query query = query().in("_id", chunkItems.toArray((String[])Array.newInstance(String.class, chunkSize)));
                 Get pageGet = new Get(query,
-                        getResponseClass(),
+                        Array.newInstance(myClass,0).getClass(),
```

Comment:
I looked up the history of this file, and noticed that this line had been incorrect for a long time (2 years). Based on that, we should've had this issue all along. How is it that it was reported as a recent regression? Did something else change that caused the regression?

Reply:
I've created `Pull` and `DeltaPull` requests 27-31.10.17. Before SDK used `Get` requests. It was updated in order to add a list of exceptions in the Response. Currently `Pull` and `DeltaPull` requests have `KinveyAbstractReadResponse` which has a `listOfException` field. But `Get` request is used inside `DeltaPull` to get items' ids for Delta Cache logic, and this Get request in `DeltaPull` has some differences in initialization in comparison with `DeltaGet` because `DeltaGet` extends `Get` but `DeltaPull` isn't.